src/tri_soft.c

```diff
@@ -682,12 +683,12 @@ static void triangle_stepper(uintptr_t state,
       }
    }
 }
-
+#include <stdio.h>
 /*
 This one will check to see what exactly we need to draw...
 I.e. this will call all of the actual renderers and set the appropriate callbacks
 */
-void _al_triangle_2d(ALLEGRO_BITMAP* texture, ALLEGRO_VERTEX* v1, ALLEGRO_VERTEX* v2, ALLEGRO_VERTEX* v3)
+void _al_triangle_2d(ALLEGRO_BITMAP* texture, ALLEGRO_VERTEX* v1, ALLEGRO_VERTEX* v2, ALLEGRO_VERTEX* v3, bool prim_addon)
```

Comment:
This is the boolean flag anti-pattern. It might be better and more performant to have two different functions for the two different cases, especially seeing we are always calling this with a true or false constant.

Reply:
I think boolean flags have their place, although in this case due to the oddity with the repeat/clamp (see the other comment), we turn out not to need this at all.